### PR TITLE
Fix various C++ warnings from LGTM

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [🤞-ish] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by kaincenteno** [comment]
what is LGTM?

**Comment by zach2good** [comment]
Code quality checker. Similar to when we used to use SonarQube, but a lot simpler: https://lgtm.com/projects/g/LandSandBoat/server/alerts/?mode=list

File: src/login/login.cpp [hunk 1]
```diff
@@ -546,7 +546,7 @@ void maint_config_read(const char* key, const char* value)
     {
         maint_config.maint_mode = std::atoi(value);
 
-        if (maint_config.maint_mode > 2 || maint_config.maint_mode < 0)
+        if (maint_config.maint_mode > 2)
```
Comment (TeoTwawki) [inline src/login/login.cpp]: guessing original implementation was copypasta, because this is only ever 0 or 1 and should have been just true/false

**Comment by zach2good** [comment]
If we have all of this cleaned up, we can probably get rid of LGTM runs in CI and adjust our build flags. Things like:
WIN:
https://github.com/LandSandBoat/server/blob/base/cmake/CompilerWarnings.cmake#L33
https://github.com/LandSandBoat/server/blob/base/cmake/CompilerWarnings.cmake#L34
https://github.com/LandSandBoat/server/blob/base/cmake/CompilerWarnings.cmake#L42

LIN:
https://github.com/LandSandBoat/server/blob/base/cmake/CompilerWarnings.cmake#L60-L61

**Comment by zach2good** [comment]
WARNING: These will make it harder to write C++, as you'll constantly be fighting to not write crappy code. This is fine for people comfortable with C++, but it might scare off or frustrate people who are just getting started.

File: src/login/login.cpp (hunk 1 above)
Comment (claywar) [reply src/login/login.cpp]: Got that covered now, good catch

**Comment by claywar** [comment]
I'm all for opening up the CI floodgates a little at a time to continue improvements.  Removed LGTM from CI due to the length of time it takes to run as well.

**Review by zach2good — APPROVED** [review]

**Pull request merged**